pkg/query/storeset.go

```diff
@@ -238,10 +239,12 @@ func (s *StoreSet) Update(ctx context.Context) {
 		// No external labels means strictly store gateway or ruler and it is fine to have access to multiple instances of them.
 		//
 		// Sidecar will error out if it will be configured with empty external labels.
-		if len(store.Labels()) > 0 && externalLabelStores[externalLabelsFromStore(store)] != 1 {
+		externalLabels := externalLabelsFromStore(store)
+		storesWithExternalLabels := externalLabelStores[externalLabels]
+		if len(store.LabelSets()) > 0 && len(store.LabelSets()[0].Labels) > 0 && storesWithExternalLabels != 1 {
```

Comment:
`len(store.LabelSets()[0].Labels) > 0 ` -> hm, what if `1` will have 0 elements? (:

I would skip this one. Let's make this natural, so we don't expect external labels to be empty? if empty the labelset is empty? What do you think?

Reply:
Yes I was thinking something similar, and with the other invariant in tsdb.go and prometheus.go this works out nicely.

Reply:
Done. It's much nicer this way.